r2/r2/controllers/front.py

```diff
@@ -874,14 +879,22 @@ def GET_about(self):
         return Reddit(content=item).render()
 
     @require_oauth2_scope("read")
-    @api_doc(api_section.subreddits, uses_site=True)
+    @api_doc(api_section.subreddits, uri='/r/{subreddit}/about/sidebar')
     def GET_sidebar(self):
         """Get the sidebar for the current subreddit"""
-        usertext = UserText(c.site, c.site.description)
-        return Reddit(content=usertext).render()
+        if isinstance(c.site, FakeSubreddit):
+            return self.abort404()
+        if is_api():
```

Comment:
I haven't checked, but that may not be the safest thing in the world since `UserText` is used elsewhere IIRC. But maybe if an empty class is made with it's parent being `UserText` and then use that, with an override on templates.py, that might be better. Don't really know. I'd have to take a closer look.